packages/sdks/javascript/src/lib/Ninetailed.ts

```diff
@@ -300,7 +299,29 @@ export class Ninetailed implements NinetailedInstance {
         }. This call will be ignored.`
       );
     } else {
-      this.observedElements.set(element, remaingPayload);
+      if (!this.observedElements.has(element)) {
+        this.observedElements.set(element, [remaingPayload]);
+      } else {
+        const existingPayloads = this.observedElements.get(element);
+
+        if (!existingPayloads) {
```

Comment:
This can never happen, as you have the `has` case before. I'd structure it a bit different as you have to do the get here anyhow, so you avoid the if / else:

```ts
const existingPayloads = this.observedElements.get(element);

if (!existingPayloads) {
  this.observedElements.set(element, [remaingPayload]);
  return;
}

... rest of the code
```